exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/otlp/traces/SpanFlags.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.exporter.internal.otlp.traces;
+
+import io.opentelemetry.api.trace.TraceFlags;
+
+/**
+ * Represents the 32 bit span flags <a
+ * href="https://github.com/open-telemetry/opentelemetry-proto/blob/342e1d4c3a1fe43312823ffb53bd38327f263059/opentelemetry/proto/trace/v1/trace.proto#L133">as
+ * specified in the proto definition</a>
+ */
+public final class SpanFlags {
+  // As defined at:
+  // https://github.com/open-telemetry/opentelemetry-proto/blob/342e1d4c3a1fe43312823ffb53bd38327f263059/opentelemetry/proto/trace/v1/trace.proto#L351-L352
+  static final int CONTEXT_HAS_IS_REMOTE_BIT = 0x00000100;
+  static final int CONTEXT_IS_REMOTE_BIT = 0x00000200;
+  static final int CONTEXT_IS_REMOTE_MASK = CONTEXT_HAS_IS_REMOTE_BIT | CONTEXT_IS_REMOTE_BIT;
+
+  private SpanFlags() {}
+
+  /**
+   * Returns the int (fixed32) representation of the {@link TraceFlags} enriched with the flags
+   * indicating a remote parent.
+   *
+   * @param isParentRemote indicates whether the parent context is remote
+   * @return the int (fixed32) representation of the {@link TraceFlags} enriched with the flags
+   *     indicating a remote parent.
+   */
+  public static int withParentIsRemoteFlags(TraceFlags traceFlags, boolean isParentRemote) {
+    byte byteRep = traceFlags.asByte();
+    if (isParentRemote) {
+      return (byteRep & 0xff) | CONTEXT_IS_REMOTE_MASK;
+    }
+    return (byteRep & 0xff) | CONTEXT_HAS_IS_REMOTE_BIT;
+  }
```

Comment:
I don't think this implementation is correct. Shouldn't we be setting the `CONTEXT_HAS_IS_REMOTE_BIT` every time this method is called, regardless of the `isParentRemote` value? I read that to be an indication that we have knowledge about our parent remote status, regardless of the status.

Also, should we be assuming that the flag is off and only toggling it on? I think we should probably turn it off when `isParentRemote == false`, right?

```suggestion
  public static int withParentIsRemoteFlags(TraceFlags traceFlags, boolean isParentRemote) {
    int byteRep = (traceFlags.asByte() | CONTEXT_HAS_IS_REMOTE_BIT) & 0xff;
    if (isParentRemote) {
      return byteRep | CONTEXT_IS_REMOTE_MASK;
    }
    return (byteRep & ~CONTEXT_IS_REMOTE_MASK) & 0xff;
  }
```

Reply:
For reference, [here's the golang implementation](https://github.com/open-telemetry/opentelemetry-go/pull/5194/files#diff-078d9913ca59000df00603baabadb474626a5e4dc6df5b8268f141f45f2968f8R163). It avoids the "turning off" thing because it's obviously off to start with. In our case, it's not as obvious.

Reply:
Oh, I totally missed the fact that there was a 3rd constant being defined above! The visual difference between `CONTEXT_IS_REMOTE_BIT` and `CONTEXT_IS_REMOTE_MASK` wasn't enough. Whoops!

You're right on both fronts -- it's correct as you have it, and the suggestion I gave would not do the right thing. Thanks for taking the time to clarify!